portfolio/src/main/webapp/script.js

```diff
@@ -64,19 +64,25 @@ function filterCourseDisplay(courseType = 'course') {
  * Fetch data from server and display on DOM.
  */
 function getAndDisplayComments() {
-  fetch('/data')
+  fetch('/list-comments')
   .then(response => response.json())
   .then((data) => {
-    const commentContainer = document.getElementById('data-container');
-    commentContainer.innerText = "";
-
-    // Display each comment on a new line.
+    // Display each comment in a block with a like button .
     for (let comment of data) {
-      commentContainer.innerText += 'Posted by: ' + comment.NAME + '\n';
-      commentContainer.innerText += comment.CONTENT + '\n';
-      commentContainer.innerText += comment.numLikes + ' likes\n';
-      commentContainer.innerText += 'Posted at ' + comment.TIMESTAMP + '\n';
-      commentContainer.innerText += '\n';
+      const commentContainer = document.createElement('div');
+      commentContainer.class = 'comment-container';
+      commentContainer.innerHTML = '';
+      commentContainer.innerHTML += 'Posted by: ' + comment.name + '<br>';
+      commentContainer.innerHTML += comment.content + '<br>';
+      commentContainer.innerHTML += comment.numLikes + ' likes<br>';
+      commentContainer.innerHTML += 'Posted at ' + comment.timestamp + '<br>';
+      // TODO: use methods to create button.
+      const likeButtonHTML = '<form action="/add-like" method="POST">' + 
+        '<input type="hidden" name="comment-key" value="' + comment.key + '" />' + 
+        '<input type="submit" value="Like" class="btn btn-danger btn-sm" /></form>';
+      commentContainer.innerHTML += likeButtonHTML;
+      commentContainer.innerHTML += '<br><br>';
+      document.getElementById('content').appendChild(commentContainer);
     }
   });
```

Comment:
Let's consider separating this function out into smaller functions - one to create the comments and one to get the comments. When creating each comment, you can do a document.createElement for each piece and put those all in the commentContainer. I think this will help clean up your code and make it more readable. You can also do this in a follow up PR :) 
Example: https://github.com/googleinterns/step/blob/master/walkthroughs/week-3-server/examples/todo-list/src/main/webapp/script.js